invenio_checks/components.py

```diff
@@ -21,31 +24,59 @@ def enabled(self):
         """Return if checks are enabled."""
         return current_app.config.get("CHECKS_ENABLED", False)
 
-    def update_draft(self, identity, data=None, record=None, errors=None, **kwargs):
+    def _run_checks(self, identity, data=None, record=None, errors=None, **kwargs):
         """Update handler."""
         errors = errors or []
 
         if not self.enabled:
             return
 
         communities = []
-        if record.review.is_open:
-            communities.append(record.review.community)
+        if record.parent.review and (
+            record.parent.review.status == "submitted"
+            or record.parent.review.status == "created"
+        ):
+            # drafts can only be submitted to one community
+            communities.append(record.parent.review.receiver.resolve().id)
+        else:
+            results = current_requests_service.search(
```

Comment:
minor: this is pretty expensive to do on every draft save (on local dev probably unnoticeable, but will hit us hard on production). I don't see a clear way around it, since we're not storing this information in the DB to be able to easily query.

So talking about solutions:

- Temporarily skip for now
- Cache a draft's requests on Redis
- Rely on something else that is query-able via the DB, e.g. creating a `CheckRun` at a specific moment, that actually has both the `record_id` and `community_id`.